src/zabapgit_object_ucsa.prog.abap

```diff
@@ -0,0 +1,251 @@
+          lx_root                   TYPE REF TO cx_root,
+          lv_text                   TYPE string,
+          lo_persistence            TYPE REF TO object,
+          lr_complete_comm_assembly TYPE REF TO data.
+
+    FIELD-SYMBOLS: <ls_complete_comm_assembly> TYPE any.
+
+    TRY.
+        CREATE DATA lr_complete_comm_assembly TYPE ('UCONSERVASCOMPLETE').
+        ASSIGN lr_complete_comm_assembly->* TO <ls_complete_comm_assembly>.
+        ASSERT sy-subrc = 0.
+
+        io_xml->read(
+          EXPORTING
+            iv_name = 'UCSA'
+          CHANGING
+            cg_data = <ls_complete_comm_assembly> ).
+
+        lv_id = ms_item-obj_name.
+
+        lo_persistence = get_persistence( lv_id ).
+
+        CALL METHOD lo_persistence->('IF_UCON_SA_PERSIST~CREATE').
+
+        CALL METHOD lo_persistence->('IF_UCON_SA_PERSIST~SAVE')
+          EXPORTING
+            sa      = <ls_complete_comm_assembly>
+            version = zif_abapgit_definitions=>gc_version-active.
+
+        tadir_insert( iv_package ).
+
+      CATCH cx_root INTO lx_root.
+        lv_text = lx_root->get_text( ).
+        zcx_abapgit_exception=>raise( lv_text ).
+    ENDTRY.
+
+  ENDMETHOD.
+
+  METHOD lif_object~delete.
+
+    DATA: lv_id          TYPE ty_id,
+          lx_root        TYPE REF TO cx_root,
+          lv_text        TYPE string,
+          lo_persistence TYPE REF TO object.
+
+    TRY.
+        lv_id = ms_item-obj_name.
+
+        lo_persistence = get_persistence( lv_id ).
+
+        CALL METHOD lo_persistence->('IF_UCON_SA_PERSIST~DELETE')
+          EXPORTING
+            version = zif_abapgit_definitions=>gc_version-active.
+
+      CATCH cx_root INTO lx_root.
+        lv_text = lx_root->get_text( ).
+        zcx_abapgit_exception=>raise( lv_text ).
+    ENDTRY.
+
+  ENDMETHOD.
+
+
+  METHOD lif_object~jump.
+
+    CALL FUNCTION 'RS_TOOL_ACCESS'
+      EXPORTING
+        operation           = 'SHOW'
+        object_name         = ms_item-obj_name
+        object_type         = ms_item-obj_type
+        in_new_window       = abap_true
+      EXCEPTIONS
+        not_executed        = 1
+        invalid_object_type = 2
+        OTHERS              = 3.
+
+    IF sy-subrc <> 0.
+      zcx_abapgit_exception=>raise( 'error from RS_TOOL_ACCESS' ).
+    ENDIF.
+
+  ENDMETHOD.
+
+  METHOD lif_object~compare_to_remote_version.
+
+    CREATE OBJECT ro_comparison_result TYPE lcl_comparison_null.
+
+  ENDMETHOD.
+
+  METHOD get_persistence.
+
+    CALL METHOD ('CL_UCON_SA_DB_PERSIST')=>('IF_UCON_SA_PERSIST~GET_INSTANCE')
+      EXPORTING
+        id       = iv_id
+      RECEIVING
+        instance = ro_persistence.
+
+  ENDMETHOD.
+
+
+  METHOD clear_dynamic_fields.
+
+    FIELD-SYMBOLS: <header> TYPE any,
+                   <field>  TYPE any.
+
+    ASSIGN COMPONENT 'HEADER' OF STRUCTURE cs_complete_comm_assembly
+           TO <header>.
+    ASSERT sy-subrc = 0.
+
+    ASSIGN COMPONENT 'CREATEDBY' OF STRUCTURE <header>
```

Comment:
Hi, @christianguenter2, looking better now.

I wonder if you could extract this into a private method like this:

METHODS clear_dynamic_field
	importing
		iv_field_name type string.

METHOD clear_dynamic_field.
FIELD-Symbols: ...

ASSIGN COMPONENT iv_fieldname OF STRUCTURE <header>
           TO <field>.
    ASSERT sy-subrc = 0.
    CLEAR: <field>.
ENDMETHOD.


Still, header should not be called is it not clean. (why?)

Reply:
Good point. I extracted a new private method.

But I don't get this one.
"Still, header should not be called is it not clean. (why?)"
What do you mean by that?